corehq/apps/app_manager/suite_xml/sections/entries.py

```diff
@@ -241,6 +242,25 @@ def include_post_in_entry(self, module_id):
         )
         return (using_inline_search or sync_on_form_entry) and not loads_registry_case
 
+    def add_post_to_entry(self, form, module, e):
+        from ..post_process.remote_requests import (
+            QuerySessionXPath,
+            RemoteRequestFactory,
+            RESULTS_INSTANCE_INLINE
+        )
+        case_session_var = self.get_case_session_var_for_form(form)
+        storage_instance = RESULTS_INSTANCE_INLINE if module_uses_inline_search(module) \
+            else 'casedb'
+        remote_request_factory = RemoteRequestFactory(
+            None, module, [], case_session_var=case_session_var, storage_instance=storage_instance,
+            exclude_relevant=case_search_sync_cases_on_form_entry_enabled_for_domain(self.app.domain))
+        e.post = remote_request_factory.build_remote_request_post()
+        if hasattr(module, 'parent_select') and module.parent_select.active:
+            data = QueryData(key='case_id')
+            data.ref = QuerySessionXPath('parent_id').instance()
```

Comment:
This may not always be `parent_id`. I'm not quite sure what the best way to figure out what the parent datum ID will be. I'm wondering if it might make sense to move this code to a post processing step at which point all the datum IDs will be known. But that might make it harder to figure out which datum's need to be included - though you could probably include all case datums.

Reply:
I see, would this work? [8e094f4](https://github.com/dimagi/commcare-hq/pull/33617/commits/8e094f4fd71d44caab36397ac1ef94df1de720a8)

I don't have a good grasp on the suite definitions and interactions. `get_datum_meta_module` seems to always return the relevant parent datum, but not sure if it'll also return datum_ids that shouldn't be added to `post`